oneoffs/joseki/joseki_query.py

```diff
@@ -0,0 +1,190 @@
+                               select next_move, sum(count) from next_moves where seq_id = ? group by 1
+                               """, (s_id,)).fetchall()
+            total = db.execute("select sum(count) from joseki_counts where seq_id = ?",
+                               (s_id,)).fetchone()[0]
+        else:
+            start = timer()
+            nexts = db.execute("""
+                               select next_move, sum(nm.count)
+                               from next_moves as nm join joseki_counts as jc
+                               on jc.id == nm.joseki_hour_id
+                               where nm.seq_id = ? and jc.run = ? group by 1
+                               """, (s_id, run)).fetchall()
+            end = timer()
+            print('%.4f seconds for fancy join.' % (end-start,))
+            total = db.execute("select sum(count) from joseki_counts where seq_id = ? and run=?",
+                               (s_id, run)).fetchone()[0]
+
+    if not nexts:
+        print("No next moves found, post params:", d['params'])
+        return flask.Response(json.dumps(res), mimetype='text/json')
+
+    next_moves = {}
+    tot = 0
+    for nxt, ct in nexts:
+        next_moves[nxt] = ct
+        tot += ct
+
+    for k in next_moves:
+        next_moves[k] /= tot
+
+    max_v = max(next_moves.values())
+
+    next_moves = {k:v / max_v for k,v in next_moves.items() if v > 0.001}
+    res = {'count': total,
+           'next_moves': next_moves}
+
+    return flask.Response(json.dumps(res), mimetype='text/json')
+
+
+@app.route("/games", methods=["POST"])
+def games():
+    d = json.loads(flask.request.data.decode('utf-8'))
+    prefix = d['params']['sgf']
+    sort_hour = d['params']['sort']
+    run = d['params']['run']
+    # "page" is 1-indexed, so subtract 1 to get the proper OFFSET.
+    page = d['params']['page'] - 1 
+    db = get_db()
+
+    if (sort_hour.lower() != 'desc' and sort_hour.lower() != 'asc'):
+        print("Invalid input for sort_hour param: ", sort_hour)
+        return flask.Response(json.dumps({'rows': []}), mimetype='text/json')
+
+    s_id = db.execute("select id from joseki where seq=?", (prefix,)).fetchone()
+    if not s_id:
+        return flask.Response(json.dumps({'rows': []}), mimetype='text/json')
+    s_id = s_id[0]
+
+    q = """select example_sgf, hour, run, b_wins*1.0/count from joseki_counts
+        where seq_id=? {} order by hour {} limit 30 offset ?""".format(
+            "and run = ?" if run else "", sort_hour)
+
+    if run:
+        rows = db.execute(q, (s_id, run, page * 30)).fetchall()
+    else:
+        rows = db.execute(q, (s_id, page * 30)).fetchall()
+    res = [ {'game': os.path.basename(r[0]), 'hour': r[1],
+             'run': r[2], 'winrate': r[3]} for r in rows]
+
+    return flask.Response(json.dumps({'rows': res}), mimetype='text/json')
+
+
+@app.route("/search", methods=["POST"])
+def search():
+    d = json.loads(flask.request.data.decode('utf-8'))
```

Comment:
if you set mimetype in the post request, then you can simply this to request.get_json()

https://stackoverflow.com/a/20001283/459714